### PR TITLE
Bug 1416347 - Implement a solution for generating a noise metric when viewing perfherder compare

### PR DESCRIPTION
**Pull request by jmaher**
this works for summary and subtest, probably has a lot of potential wins and logic to improve upon- for now I want to get a first attempt rolled out for easier comparison of new platforms.

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

File: ui/partials/perf/comparesubtestctrl.html [hunk 1]
```diff
@@ -7,6 +7,16 @@
   <div id="error" ng-if="!dataLoading && errors.length">
     <compare-error errors="errors" original-revision="originalRevision" original-project="originalProject" new-revision="newRevision" new-project="newProject"></compare-error>
   </div>
+    <div class="alert alert-warning" role="alert" ng-if="testsTooVariable.length > 1 && filterOptions.showOnlyNoise">
```
Comment (wlach) [inline ui/partials/perf/comparesubtestctrl.html]: I don't think you need the `> 1` here.

File: ui/js/controllers/perf/compare.js [hunk 2]
```diff
@@ -149,11 +160,21 @@ perf.controller('CompareResultsCtrl', [
...
                     if (cmap.isEmpty) {
                         return;
                     }
 
+                    // TODO: ideally anything >10.0 is bad, but should we ignore anything?
+                    if (cmap.originalStddevPct < 50.0 && cmap.newStddevPct < 50.0) {
+                        $scope.oldStddevVariance[platform].values.push(Math.round(cmap.originalStddevPct * 100) / 100);
+                        $scope.newStddevVariance[platform].values.push(Math.round(cmap.newStddevPct * 100) / 100);
+                    } else if (cmap.originalStddevPct === undefined || cmap.newStddevPct === undefined) {
+                        // TODO: figure out why this is the case
```
Comment (wlach) [inline ui/js/controllers/perf/compare.js]: is this still TODO? the line just below implies not

File: ui/js/controllers/perf/compare.js [hunk 3]
```diff
@@ -149,11 +160,21 @@ perf.controller('CompareResultsCtrl', [
...
+                        // No results for one or both data points
+                    } else {
+                        // TODO: this is sloppy in the UI, but gives good information
```
Comment (wlach) [inline ui/js/controllers/perf/compare.js]: how would you do it differently? I think this is probably ok.

File: ui/partials/perf/comparesubtestctrl.html (hunk 1 above)
Comment (jmaher) [reply ui/partials/perf/comparesubtestctrl.html]: the 1st entry in the array is the hacky title row, so I want to ignore that if there are no tests we needed to ignore- the rest of the suggestions are great and make sense.

**Pull request merged**